### DISCUSSION TITLE
Some saved searches that might come in handy (looking for problems)

### DISCUSSION THREAD

**Discussion by Moonbase59** [quodlibet/quodlibet#4349] [Show and tell]
Just sharing a few saved searches I use to find problems in my collection:

#### - BPM 0
```
#(bpm<1)
```

#### - BPM missing
```
bpm=""
```

#### - Cover image missing
```
~picture=""
```

#### - Exotic formats (not FLAC, MP3)
```
!~format=|(flac,mp3)
```
(You could add "ogg" to that—it is not an exotic format at all. :grin:)

#### - Filename potentially problematic (characters)
```
~filename=/[\\?%*:|"<>\x7F\x00-\x1F]/
```
This shows tracks with a path/filename that is perfectly legal on Linux but could be problematic for other operating systems like Windows or MacOS, because it contains characters "illegal" for those.

#### - Filename potentially problematic (length)
```
~filename=!/^.{1,255}$/
```
This shows tracks whose path/filename might be too long for Windows systems.

#### - language missing
```
language=""
```

#### - Modified within last week
```
#(mtime < 1 week)
```

#### - ReplayGain (Track) missing
```
&(~format=!wav,replaygain_track_gain="")
```

#### Test: "a/b"? 'ÄÖÜẞäöüß 测试 テスト Тест! *%<> \hello
```
"test"
```
This is for testing playlist exporters—they must be able to generate a working playlist from that saved search (by sanitizing the file name).

The reason that all these "technical" saved searches start with a dash ("-") is that my `ql-playlists` fetches the list of saved searches from QL and produces an M3U playlist for each one _except those beginning with a dash_.

**Comment by Npa972** [comment]
Thanks you for ~picture="" (and others examples).
